test/cluster/mzcompose.py

```diff
@@ -4156,6 +4156,67 @@ def check_introspection():
         check_introspection()
 
 
+def workflow_test_github_8734(c: Composition) -> None:
+    """
+    Tests that REFRESH MVs on paused clusters don't unnecessarily hold back
+    compaction of their inputs after an envd restart.
+
+    Regression test for database-issues#8734.
+    """
+
+    def check_read_frontier_advances():
```

Comment:
There is a similar function in the refresh restart test: https://github.com/MaterializeInc/materialize/blob/b215eaa5697c6be4035c7e1785ca99672061687d/test/cluster/mzcompose.py#L3738-L3749
It would be good to factor these out into a global function.

Reply:
Btw., in theory, `workflow_test_github_8734` could fit into `workflow_test_refresh_mv_restart`. But that test is already quite complicated, so I can totally understand if you'd like to avoid putting this new test there. But you could maybe add a comment to `workflow_test_refresh_mv_restart` that there are more refresh tests involving restarts in `workflow_test_github_8734` (and also in `workflow_test_refresh_mv_warmup`).

Reply:
Thx, done and done!

And yeah, I briefly considered adding this to `workflow_test_refresh_mv_restart` but then I got too confused and gave up 😅